### DISCUSSION TITLE
How can I keep {{ }} in stencil?

### DISCUSSION THREAD

**Discussion by wulinman** [SwiftGen/StencilSwiftKit#172] [Q&A] [answered]
want a output with {{ }} like {{\(key)}}, how can I keep {{ }} ?
Example: 

args.forEach { key, value in
    result = result.replacingOccurrences(of: "{{\(key)}}", with: value)
}

but output comes as:
args.forEach { key, value in
    result = result.replacingOccurrences(of: "", with: value)
}

**Comment by djbe** [comment, marked as answer]
In Stencil, `{{ stuff }}` refers to a variable. In your case you're referring to the variable named `(key)`.

Stencil is based on the templating language in Django, which has [the same issue](https://stackoverflow.com/questions/7772001/how-to-escape-or-in-django-template). Note that Stencil does **not** support every feature that Django supports.

Anyway, quick solution is `{{ '{' }}` to print a `{` character. Stencil even has a test for this "case": https://github.com/stencilproject/Stencil/blob/master/Tests/StencilTests/LexerSpec.swift#L129-L140
